src/WebJobs.Script.WebHost/Diagnostics/DiagnosticLogger.cs

```diff
@@ -0,0 +1,88 @@
+        };
+
+        private readonly string _category;
+        private readonly IEventGenerator _eventGenerator;
+        private readonly IEnvironment _environment;
+        private readonly string _websiteHostName;
+
+        public DiagnosticLogger(string category, IEventGenerator eventGenerator, IEnvironment environment, IExternalScopeProvider scopeProvider)
+        {
+            _category = category ?? throw new ArgumentNullException(nameof(category));
+            _eventGenerator = eventGenerator ?? throw new ArgumentNullException(nameof(eventGenerator));
+            _environment = environment ?? throw new ArgumentNullException(nameof(environment));
+            _scopeProvider = scopeProvider ?? throw new ArgumentNullException(nameof(scopeProvider));
+
+            _regionName = _environment.GetEnvironmentVariable(EnvironmentSettingNames.RegionName) ?? string.Empty;
+            _websiteHostName = _environment.GetEnvironmentVariable(EnvironmentSettingNames.AzureWebsiteHostName);
+        }
+
+        public IDisposable BeginScope<TState>(TState state) => _scopeProvider.Push(state);
+
+        public bool IsEnabled(LogLevel logLevel)
+        {
+            return true;
+        }
+
+        public void Log<TState>(LogLevel logLevel, EventId eventId, TState state, Exception exception, Func<TState, Exception, string> formatter)
+        {
+            if (!IsEnabled(logLevel))
+            {
+                return;
+            }
+
+            string formattedMessage = formatter?.Invoke(state, exception);
+
+            // Make sure we have something to log
+            if (string.IsNullOrEmpty(formattedMessage) && exception == null)
+            {
+                return;
+            }
+
+            (string exceptionType, string exceptionMessage, string exceptionDetails) = exception.GetExceptionDetails();
+
+            var scopeProps = _scopeProvider.GetScopeDictionary();
+
+            // Build up a JSON string for the Azure Monitor 'properties' bag
+            // Any null values will be dropped from the resulting JSON string
+            var properties = new
+            {
+                exceptionType,
+                exceptionMessage,
+                exceptionDetails,
+                message = Sanitizer.Sanitize(formattedMessage),
+                category = _category,
+                hostVersion = _hostVersion,
+                functionInvocationId = Utility.GetValueFromScope(scopeProps, ScopeKeys.FunctionInvocationId),
+                functionName = Utility.GetValueFromScope(scopeProps, ScopeKeys.FunctionName),
+                hostInstanceId = Utility.GetValueFromScope(scopeProps, ScopeKeys.HostInstanceId),
+                activityId = Utility.GetValueFromScope(scopeProps, ScriptConstants.LogPropertyActivityIdKey),
+                level = logLevel.ToString()
+            };
+
+            string jsonString = JsonConvert.SerializeObject(properties, Formatting.None, _serializerSettings);
```

Comment:
Rather than serialize here, I'll change this to construct the JObject explicitly to save the reflection cost.